### ISSUE TITLE
When updating a company through the identify method only a subset of the semantic traits are conserved

### ISSUE THREAD

**Issue by sboichon** [segment-integrations/integration-intercom#34]
Hi guys !
Because of #33, we had to start using identify to update companies in intercom.
Problem is the [formatCompany]( https://github.com/segment-integrations/integration-intercom/blob/master/lib/mapper.js#L194-L219) function is only passing a subset of the company semantics attributes. 
This results in intercom receiving reserved attributes in custom attributes and us having duplicates attributes with the same name.
Is there a reason for the logic of formatCompany to be different than the one from [mapper](https://github.com/segment-integrations/integration-intercom/blob/master/lib/mapper.js#L138-L182)

Thanks !

**Comment by hankim813** [comment]
@sboichon thanks yep we can fix that. I'll try to ship a PR this week for this issue!

**Comment by sboichon** [comment]
Thanks @hankim813 !
I'm not a fierce competitor, so I'll let you win this battle ! (And I actually have already plenty on my plate !)

**Comment by hankim813** [comment]
@sboichon fyi -- this new intercom version only affects those who created a new source and enabled intercom _after_ we merged the PR. If you had intercom set up through segment for a while before the merge, you should not be affected by this update.

**Comment by hankim813** [comment]
If you give me your sourceId from the settings, I can confirm for you which version you are definitively on :) @sboichon

**Comment by marcboscher** [comment]
Hi @hankim813 , I'm a colleague of @sboichon 
Our source ids are bB2qAGTZe2 and 4CvAhTAbnh (I assume this is not sensitive?)

Could you tell us which version of the integration we're running?

The fact that releases to this repo don't automatically update our integration would explain A LOT of things. For example, Intercom has checked their logs and clearly calls to group() still call the /user endpoint, not the /bulk endpoint (which was changed in version 2.0.0)

How do we force an updated of the code?

**Comment by hankim813** [comment]
@marcboscher sorry about the confusion... I know it's not ideal. Intercom is a special integration for us since it's one of our most used one and shipping changes can be tricky since we can't easily push down breaking changes to all our mutual customers.

With that said, I've confirmed on our end that you are on the v1 version (can be seen in `2016-10-26` branch).

The new version is the default for all users that created a new source and enabled intercom after **Nov 11th**. If you had sources set up with Intercom prior, you will be on the V1 version. 

We plan on migrating all our customers in the coming weeks. If you'd like to be upgraded right now, you can send me an email and ask! I need the sourceIds where you want the new intercom version. Please send the email with the same email you signed up on segment.com for verification!

**Comment by sboichon** [comment]
@hankim813 Hi !
Thanks for the fix.
Could it be possible to migrated on the v2 for our test environment only at first and to be able to synchronise with you guys for the migration of our prod environment ?
Maybe we can continue this by mail: sylvainb@unito.io

**Comment by hankim813** [comment]
@sboichon ahh feel free to just create a new staging source in Segment (if you don't already have one, it's recommended). If you do that and enable intercom it should be v2 by default and you can hook it all up to your test environment/test intercom accounts.

Once you are happy, we can do the migration :)

**Comment by brennan** [comment]
Hey @hankim813, chiming in here. Looks like @sboichon is having the same issue surfaced back on Nov. 9 - server-side, we're sending semantic company attributes to Intercom as "custom_attributes" on the company object.

**Comment by sboichon** [comment]
Just updated identify-company-v1.json fixtures with : 
```
{
  "input": {
    "type": "identify",
    "userId": "user-id",
    "timestamp": "2014",
    "traits": {
      "company": "Segment",
      "plan": "paid",
      "monthlySpend": 1239
    }
  },
  "output": {
    "user_id": "user-id",
    "last_request_at": 1388534400,
    "custom_attributes": {
      "id": "user-id"
    },
    "companies": [{
      "company_id": 2017828326,
      "custom_attributes": {},
      "name": "Segment",
      "plan": "paid",
      "monthlySpend": 1239
    }]
  }
}
```
test don't pass.

**Comment by hankim813** [comment]
@sboichon @brennan thanks guys -- everyone is back on the v1 as per Intercom's request. When we initially shipped #35, which was the fix to this issue, it was only updated for v2. We will ship the same fix against v1.

**Comment by hankim813** [comment]
this should be fixed! please let me know if you still have issues
